geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -187,21 +187,22 @@ long zadd(Region<RedisKey, RedisData> region, RedisKey key, List<byte[]> members
       ZAddOptions options) {
     AddsDeltaInfo deltaInfo = null;
     Iterator<byte[]> iterator = membersToAdd.iterator();
-
     int initialSize = getSortedSetSize();
-
+    int changesCount = 0;
     while (iterator.hasNext()) {
       byte[] score = iterator.next();
       validateScoreIsDouble(score);
       byte[] member = iterator.next();
-
       if (options.isNX() && members.containsKey(member)) {
         continue;
       }
       if (options.isXX() && !members.containsKey(member)) {
         continue;
       }
-      memberAdd(member, score);
+      byte[] oldScore = memberAdd(member, score);
+      if (options.isCH() && oldScore != null) {
+        changesCount = getChangesCount(changesCount, oldScore, score);
+      }
```

Comment:
This can be simplified to:
```
      if (options.isCH() && oldScore != null && !Arrays.equals(oldScore, score)) {
        changesCount++;
      }
```